docs/manual-CN/5.appendix/gremlin-ngql.md

```diff
@@ -92,125 +92,524 @@ gremlin> g.V(jesus).property('age', 6000);
+----------------------------------------------------------------------------------------------------------------
+| 0           | 0                    | 0                    | 0           | 4594048193862126013 | 0            |
+----------------------------------------------------------------------------------------------------------------
+
+# 查询顶点 hercules 到顶点 cerberus 之间的路径
+# 循环的终止条件是遇到名称是 cerberus 的顶点
+gremlin> g.V().hasLabel('character').has('name','hercules').repeat(out()).until(has('name', 'cerberus')).path();
+==>[v[6],v[11]]
+==>[v[6],v[2],v[8],v[11]]
+==>[v[6],v[2],v[5],v[8],v[11]]
+...
+nebula> # Coming soon
+
+# 查询点 hercules 的所有出边可到达点的路径
+# 且终点必须是 character 类型的点
+gremlin> g.V().hasLabel('character').has('name','hercules').repeat(out()).emit(hasLabel('character')).path();
+==>[v[6],v[7]]
+==>[v[6],v[2]]
+==>[v[6],v[9]]
+==>[v[6],v[10]]
+...
+nebula> # Coming soon
+
+# 查询两顶点 pluto 和 saturn 之间的最短路径
+# 且最大深度为 3
+gremlin> g.V('pluto').repeat(out().simplePath()).until(hasId('saturn').and().loops().is(lte(3))).hasId('saturn').path();
+nebula> FIND SHORTEST PATH FROM hash('pluto') TO hash('saturn') OVER * UPTO 3 STEPS;
+=================================================================================================
+| _path_              |
+=================================================================================================
+| 6761447489613431910 <brother,0> 4863977009196259577 <father,0> -4316810810681305233
+-------------------------------------------------------------------------------------------------
+```
+
+### 查询结果排序
+
+名称               | Gremlin | nGQL           |
+-----              |---------|   -----       |
+升序排列 | order().by()    | ORDER BY         |
+降序排列 | order().by(decr) | ORDER BY DESC |
+随机排列 | order().by(shuffle) | \ |
+
+```bash
+# 查询 pluto 的兄弟并按照年龄降序排列
+gremlin> g.V(pluto).out('brother').order().by('age', decr).valueMap();
+==>[name:[jupiter],type:[god],age:[5000]]
+==>[name:[neptune],type:[god],age:[4500]]
+nebula> GO FROM hash('pluto') OVER brother YIELD $$.character.name AS Name, $$.character.age as Age | ORDER BY Age DESC;
+==================
+| Name    | Age  |
+==================
+| jupiter | 5000 |
+------------------
+| neptune | 4500 |
+------------------
+```
+
+### Group By
+
+名称               | Gremlin | nGQL           |
+-----              |---------|   -----       |
+对结果集进行分组 | group().by()    | GROUP BY         |
+去除相同元素 | dedup() | \ |
+对结果集进行分组并统计 | groupCount() | GROUP BY COUNT  |
+
+**注意：** GROUP BY 函数只能与 YIELD 语句一起使用。
+
+```bash
+# 根据顶点类别进行分组并统计各个类别的数量
+gremlin> g.V().group().by(label).by(count());
+==>[character:9,location:3]
+nebula> # Coming soon
+
+# 查询点 jupiter 出边邻点，使用 name 分组并统计
+gremlin> g.V(jupiter).out().group().by('name').by(count());
+==>[sky:1,saturn:1,neptune:1,pluto:1]
+nebula> GO FROM hash('jupiter') OVER * YIELD $$.character.name AS Name, $$.character.age as Age, $$.location.name | \
```

Comment:
@CPWstatic  I'm not sure about the count(*).  can you check it again?

Reply:
I tested every query, they all work well.

Reply:
Supported.